near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs

```diff
@@ -101,3 +109,103 @@ impl ParsedVAA {
         }
     }
 }
+
+sol! {
+    struct InitTransferWh {
+        string token;
+        uint128 amount;
+        uint128 fee;
+        string recipient;
+        address sender;
+    }
+
+    struct FinTransferWh {
+        string token;
+        uint128 amount;
+        string recipient;
+        uint128 nonce;
+    }
+
+    struct DeployTokenWh {
+        string token;
+        address tokenAddress;
+    }
+}
+
+impl TryInto<InitTransferMessage> for ParsedVAA {
+    type Error = String;
+
+    fn try_into(self) -> Result<InitTransferMessage, String> {
+        let data: &[u8] = &self.payload[1..];
+        let transfer = InitTransferWh::abi_decode(data, true).map_err(stringify)?;
+
+        Ok(InitTransferMessage {
+            transfer: TransferMessage {
+                token: transfer.token.parse().map_err(stringify)?,
+                amount: transfer.amount.into(),
+                fee: transfer.fee.into(),
+                recipient: transfer.recipient.parse().map_err(stringify)?,
+                origin_nonce: 0.into(),
```

Comment:
The nonce has been added here 
https://github.com/Near-One/omni-bridge/pull/32/files#diff-a58607475fb9a2fd5a61cb26a07165d31be15e2d05eb112c95be029d3cc68c5fR71

Reply:
556dc8201a0932aacc8052dd7f7f9dac27e4493f